src/datajudge/constraints/stats.py

```diff
@@ -0,0 +1,57 @@
+from typing import Any, Collection, Optional, Tuple
+
+import sqlalchemy as sa
+
+from .. import db_access
+from ..db_access import DataReference
+from .base import Constraint, OptionalSelections
+
+
+class KolmogorovSmirnov2Sample(Constraint):
+    def __init__(
+        self, ref: DataReference, ref2: DataReference, significance_level: float = 0.05
+    ):
+        self.significance_level = significance_level
+        super().__init__(ref, ref2=ref2)
+
+    @staticmethod
+    def calculate_2sample_ks_test(data: Collection, data2: Collection) -> float:
+        """
+        For two given lists of values calculates the Kolmogorov-Smirnov test.
+        Read more here: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.kstest.html
+        """
+        try:
+            from scipy.stats import ks_2samp
```

Comment:
I don't perfectly understand why we need this even though `scipy` is listed as dependency in `pyproject.toml`. However, if you feel confident about this, please go ahead.

Reply:
You're right @kklein , however, the reason why `scipy` is in the `pyproject.toml` file is due to read the docs. Don't ask me why it was failing, but after adding `scipy` here it worked again.

If that weren't the case, this piece of code would be useful to manage the message the user encounters when trying to use this constraint while not having `scipy` installed.